lib/task_bunny/publisher.ex

```diff
@@ -31,15 +34,16 @@ defmodule TaskBunny.Publisher do
     #{host}:#{queue}: #{inspect(message)}. options = #{inspect(options)}
     """)
 
-    conn = TaskBunny.Connection.get_connection!(host)
-
     exchange = ""
     routing_key = queue
     options = Keyword.merge([persistent: true], options)
 
-    with {:ok, channel} <- AMQP.Channel.open(conn),
-         :ok <- AMQP.Basic.publish(channel, exchange, routing_key, message, options),
-         :ok <- AMQP.Channel.close(channel) do
+    with :ok <-
+           :poolboy.transaction(
+             :publisher,
+             &GenServer.call(&1, {:publish, host, exchange, routing_key, message, options}),
+             @timeout
+           ) do
```

Comment:
It now doesn't really make sense to use with clause here. Can you change it to case or if?